tests/FeeSharingCollectorTest.js

```diff
@@ -291,21 +288,508 @@ contract("FeeSharingCollectorProxy:", (accounts) => {
+                account1,
+                SOVToken.address
+            );
+
+            let tx = await feeSharingCollector.withdrawStartingFromCheckpoint(
+                SOVToken.address,
+                nextPositive.checkpointNum.toNumber(),
+                3,
+                ZERO_ADDRESS,
+                { from: account1 }
+            );
+
+            expectEvent(tx, "UserFeeWithdrawn", {
+                sender: account1,
+                token: SOVToken.address,
+                amount: new BN(60),
+            });
+
+            let processedCheckpoints = await feeSharingCollector.processedCheckpoints.call(
+                account1,
+                SOVToken.address
+            );
+            expect(processedCheckpoints.toNumber()).to.equal(10);
+        });
+        it("withdrawRBTCStartingFromCheckpoint calculates fees correctly", async () => {
+            // To test this, create 9 checkpoints while the user has no stake, then stake with the user, create another checkpoint and call withdrawStartingFromCheckpoint with _fromCheckpoint = 10  and _maxCheckpoints = 3
+
+            /// RBTC
+            await stake(900, root);
+            const userStake = 100;
+
+            await SOVToken.transfer(account1, userStake);
+            await createCheckpoints(9);
+
+            await stake(userStake, account1);
+            await createCheckpoints(1);
+
+            let nextPositive = await feeSharingCollector.getNextPositiveUserCheckpoint(
+                account1,
+                RBTC_DUMMY_ADDRESS_FOR_CHECKPOINT
+            );
+
+            let tx = await feeSharingCollector.withdrawRBTCStartingFromCheckpoint(
+                nextPositive.checkpointNum.toNumber(),
+                3,
+                ZERO_ADDRESS,
+                { from: account1 }
+            );
+
+            expectEvent(tx, "RBTCWithdrawn", {
+                sender: account1,
+                amount: new BN(60),
+            });
+
+            let processedCheckpoints = await feeSharingCollector.processedCheckpoints.call(
+                account1,
+                RBTC_DUMMY_ADDRESS_FOR_CHECKPOINT
+            );
+            expect(processedCheckpoints.toNumber()).to.equal(10);
+
+            /// SOV
+            //mock data
+            let lendingFeeTokensHeld = new BN(wei("1", "ether"));
+            let tradingFeeTokensHeld = new BN(wei("2", "ether"));
+            let borrowingFeeTokensHeld = new BN(wei("3", "ether"));
+            let totalFeeTokensHeld = lendingFeeTokensHeld
+                .add(tradingFeeTokensHeld)
+                .add(borrowingFeeTokensHeld);
+            let feeAmount = await setFeeTokensHeld(
+                lendingFeeTokensHeld,
+                tradingFeeTokensHeld,
+                borrowingFeeTokensHeld,
+                false,
+                true
+            );
+
+            tx = await feeSharingCollector.withdrawFees([SOVToken.address]);
```

Comment:
No need to do assignment here (tx = ).
And seems like it's missing another check after .withdrawFeess()

Reply:
a copypaste lefover    
it is covered in the previous test  
removing it

Reply:
a copypaste leftover  
removing it